src/test/java/org/assertj/core/test/TypeCanonizer.java

```diff
@@ -0,0 +1,127 @@
+/**
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ * <p>
+ * Copyright 2012-2017 the original author or authors.
+ */
+package org.assertj.core.test;
+
+import com.google.common.reflect.TypeResolver;
+
+import java.io.InputStream;
+import java.lang.reflect.GenericArrayType;
+import java.lang.reflect.ParameterizedType;
+import java.lang.reflect.Type;
+import java.lang.reflect.TypeVariable;
+import java.lang.reflect.WildcardType;
+import java.util.ArrayDeque;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Queue;
+import java.util.Set;
+
+/**
+ * A Type canonizar that helps with the resolving of a {@link Type} so it can be compared to a similar one
+ * considering generics.
+ * @author Filip Hrisafov
+ * @author Clement Mathieu
+ */
+public class TypeCanonizer {
```

Comment:
I am not sure if the other AssertJ projects need something like this. If they do then maybe we should add this to the parent pom so everyone can use it